agent/utils/register.go

```diff
@@ -45,17 +46,33 @@ func (r *Reg) Add(key keyDID, value ...string) {
 	r.r[key] = value
 }
 
-func (r *Reg) Load(filename string) error {
+func initFile(filename string) (err error) {
+	if filename == "" {
+		return
+	}
+	_, err = os.Stat(filename)
+	if os.IsNotExist(err) {
+		err = writeJSONFile(filename, []byte("{}"))
+	}
+	return
+}
+
+func (r *Reg) Load(filename string) (err error) {
+	defer err2.Return(&err)
```

Comment:
One question about this @techlab-lainio: is it safe to use both `defer Unlock()` and `defer err2.Return` in same function?

Reply:
Yes, it is safe. You can add as many deferred functions into function as is needed. They all are then called in opposite order when the function is left. Because `err2.Return()` doesn't do anything else but check the error value, it doesn't affect other deferred functions. And usually deferred functions don't affect each other, but if you use **closures** you can have side-effects with function variables because they can be accessed freely. `err2` package only accesses the `err` variable and thru the pointer, and btw this is the reason it must be a pointer with named functions (like `err2`) and not closures.

Reply:
Thanks! I thought so, but no harm double-checking 😉

Reply:
Exactly! And it isn't so obvious. Go's `defer` is scoped to function not to block, i.e. lexically. In this document there is a good explanation: 
> The current draft design makes error handlers chain lexically, while `defer` builds up a chain at runtime depending on what code executes.